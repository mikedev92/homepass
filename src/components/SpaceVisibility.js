// @flow
import React from 'react';
import {
  Container,
  Header,
  Button,
  Icon,
  Body,
  Title,
  Separator,
  List,
  ListItem,
  Text,
  Left,
  Right,
  Content,
  Radio,
} from 'native-base';
import { compose } from 'recompose';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

type Props = {
  navigation: Navigation,
  data: ApolloData<Space>,
  setVisibility: string => void => void,
};

function SpaceVisibility({ navigation, data, setVisibility }: Props) {
  const visibility = data.node.visibility;
  return (
    <Container backgroundDark>
      <Header bordered>
        <Left>
          <Button onPress={() => navigation.goBack()} transparent primary>
            <Icon name="arrow-back" />
          </Button>
        </Left>
        <Body><Title>Visibility</Title></Body>
        <Right />
      </Header>
      <Content>
        <Separator bordered noTopBorder />
        <List>
          <ListItem onPress={setVisibility('OPEN')}>
            <Text>
              Open{'\n'}
              <Text caption>
                Non members can join.
              </Text>
            </Text>
            <Right>
              <Radio selected={visibility === 'OPEN'} />
            </Right>
          </ListItem>
          <ListItem onPress={setVisibility('PRIVATE')}>
            <Text>
              Private{'\n'}
              <Text caption>
                Non members must request to join.
              </Text>
            </Text>
            <Right>
              <Radio selected={visibility === 'PRIVATE'} />
            </Right>
          </ListItem>
          <ListItem last onPress={setVisibility('SECRET')}>
            <Text>
              Secret{'\n'}
              <Text caption>
                Only members can find and must be manually added.
              </Text>
            </Text>
            <Right>
              <Radio selected={visibility === 'SECRET'} />
            </Right>
          </ListItem>
        </List>
      </Content>
    </Container>
  );
}

const SpaceVisibilityQuery = gql`
  query SpaceVisibilityQuery($id: ID!) {
    node(id: $id) {
      id
      ... on Space {
        spaceId
        visibility
      }
    }
  }
`;

const SetVisibilityMutation = gql`
  mutation($input: SetSpaceVisibilityInput!) {
    setSpaceVisibility(input: $input) {
      space {
        id
        visibility
      }
    }
  }
`;

export default compose(
  graphql(SpaceVisibilityQuery, {
    options: props => ({ variables: { id: props.navigation.state.params.id } }),
  }),
  graphql(SetVisibilityMutation, {
    props: ({ mutate, ownProps }) => ({
      setVisibility: visibility => () =>
        mutate({
          variables: { input: { visibility, spaceId: ownProps.data.node.spaceId } },
          optimisticResponse: {
            __typename: 'Mutation',
            setSpaceVisibility: {
              __typename: 'SetSpaceVisibilityPayload',
              space: {
                __typename: 'Space',
                id: ownProps.data.node.id,
                visibility,
              },
            },
          },
        }),
    }),
  }),
)(SpaceVisibility);
