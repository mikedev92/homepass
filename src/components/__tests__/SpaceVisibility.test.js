// @flow
import React from 'react';
import { shallow } from 'enzyme';
import SpaceVisibility from '../SpaceVisibility';

const testProps = [
  {
    data: {
      node: {
        visibility: 'OPEN',
      },
    },
  },
  {
    data: {
      node: {
        visibility: 'PRIVATE',
      },
    },
  },
  {
    data: {
      node: {
        visibility: 'SECRET',
      },
    },
  },
];

describe('<SpaceVisibility />', () => {
  it('matches snapshots', () => {
    testProps.forEach((props) => {
      const wrapper = shallow(<SpaceVisibility setVisibility={jest.fn()} {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
