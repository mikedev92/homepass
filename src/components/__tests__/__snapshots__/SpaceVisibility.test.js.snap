// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpaceVisibility /> matches snapshots 1`] = `
<Styled(Container)
  backgroundDark={true}
>
  <Styled(Header)
    bordered={true}
  >
    <Styled(Left)>
      <Styled(Button)
        onPress={[Function]}
        primary={true}
        transparent={true}
      >
        <Styled(Icon)
          name="arrow-back"
        />
      </Styled(Button)>
    </Styled(Left)>
    <Styled(Body)>
      <Styled(Title)>
        Visibility
      </Styled(Title)>
    </Styled(Body)>
    <Styled(Right) />
  </Styled(Header)>
  <Styled(Content)>
    <Styled(Separator)
      bordered={true}
      noTopBorder={true}
    />
    <Styled(List)>
      <Styled(ListItem)>
        <Styled(Text)>
          Open
          
          
          <Styled(Text)
            caption={true}
          >
            Non members can join.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={true}
          />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)>
        <Styled(Text)>
          Private
          
          
          <Styled(Text)
            caption={true}
          >
            Non members must request to join.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={false}
          />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)
        last={true}
      >
        <Styled(Text)>
          Secret
          
          
          <Styled(Text)
            caption={true}
          >
            Only members can find and must be manually added.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={false}
          />
        </Styled(Right)>
      </Styled(ListItem)>
    </Styled(List)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`<SpaceVisibility /> matches snapshots 2`] = `
<Styled(Container)
  backgroundDark={true}
>
  <Styled(Header)
    bordered={true}
  >
    <Styled(Left)>
      <Styled(Button)
        onPress={[Function]}
        primary={true}
        transparent={true}
      >
        <Styled(Icon)
          name="arrow-back"
        />
      </Styled(Button)>
    </Styled(Left)>
    <Styled(Body)>
      <Styled(Title)>
        Visibility
      </Styled(Title)>
    </Styled(Body)>
    <Styled(Right) />
  </Styled(Header)>
  <Styled(Content)>
    <Styled(Separator)
      bordered={true}
      noTopBorder={true}
    />
    <Styled(List)>
      <Styled(ListItem)>
        <Styled(Text)>
          Open
          
          
          <Styled(Text)
            caption={true}
          >
            Non members can join.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={false}
          />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)>
        <Styled(Text)>
          Private
          
          
          <Styled(Text)
            caption={true}
          >
            Non members must request to join.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={true}
          />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)
        last={true}
      >
        <Styled(Text)>
          Secret
          
          
          <Styled(Text)
            caption={true}
          >
            Only members can find and must be manually added.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={false}
          />
        </Styled(Right)>
      </Styled(ListItem)>
    </Styled(List)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`<SpaceVisibility /> matches snapshots 3`] = `
<Styled(Container)
  backgroundDark={true}
>
  <Styled(Header)
    bordered={true}
  >
    <Styled(Left)>
      <Styled(Button)
        onPress={[Function]}
        primary={true}
        transparent={true}
      >
        <Styled(Icon)
          name="arrow-back"
        />
      </Styled(Button)>
    </Styled(Left)>
    <Styled(Body)>
      <Styled(Title)>
        Visibility
      </Styled(Title)>
    </Styled(Body)>
    <Styled(Right) />
  </Styled(Header)>
  <Styled(Content)>
    <Styled(Separator)
      bordered={true}
      noTopBorder={true}
    />
    <Styled(List)>
      <Styled(ListItem)>
        <Styled(Text)>
          Open
          
          
          <Styled(Text)
            caption={true}
          >
            Non members can join.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={false}
          />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)>
        <Styled(Text)>
          Private
          
          
          <Styled(Text)
            caption={true}
          >
            Non members must request to join.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={false}
          />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)
        last={true}
      >
        <Styled(Text)>
          Secret
          
          
          <Styled(Text)
            caption={true}
          >
            Only members can find and must be manually added.
          </Styled(Text)>
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={true}
          />
        </Styled(Right)>
      </Styled(ListItem)>
    </Styled(List)>
  </Styled(Content)>
</Styled(Container)>
`;
